cpp/tests/utilities/base_fixture.hpp

```diff
@@ -180,19 +180,16 @@ class TempDirTestEnvironment : public ::testing::Environment {
  *        Accepted types are "pool", "cuda", and "managed" only.
  * @return Memory resource instance
  */
-inline auto create_memory_resource(std::string const& allocation_mode)
+inline std::unique_ptr<rmm::mr::device_memory_resource> create_memory_resource(std::string const& allocation_mode)
 {
     std::unique_ptr<rmm::mr::device_memory_resource> rmm_resource{};
```

Comment:
```suggestion
```
I think this and the trailing return can be removed now since the default will throw anyway